### ISSUE TITLE
Add an 'All Attributes' constructor to ColorScheme

### ISSUE THREAD

**Issue by tznind** [gui-cs/Terminal.Gui#3345]
**Is your feature request related to a problem? Please describe.**
I have started trying to [upgrade TerminalGuiDesigner to use the latest nuget package](https://github.com/gui-cs/TerminalGuiDesigner/pull/289).  A lot has changed since 2.0.0-pre.250!

One thing I am trying to tackle is the new `init` only properties.

Code DOM does not support initializer syntax (see [this stack overflow post])(https://stackoverflow.com/a/12874957/4824531).  Post is old but I have checked latest prerelease nuget packages and still do not see any option for object initializers.

![image](https://github.com/gui-cs/Terminal.Gui/assets/31306100/53cf1c7f-34ad-41e8-b678-a10ae0c5d129)

**Describe the solution you'd like**

Can we add a constructor that takes the relevant properties as parameters?

Example of current code produced by designer:

```csharp
     this.dialogBackground = new Terminal.Gui.ColorScheme();
     this.dialogBackground.Normal = new Terminal.Gui.Attribute(Terminal.Gui.Color.White, Terminal.Gui.Color.DarkGray);
     this.dialogBackground.HotNormal = new Terminal.Gui.Attribute(Terminal.Gui.Color.White, Terminal.Gui.Color.DarkGray);
     this.dialogBackground.Focus = new Terminal.Gui.Attribute(Terminal.Gui.Color.White, Terminal.Gui.Color.DarkGray);
     this.dialogBackground.HotFocus = new Terminal.Gui.Attribute(Terminal.Gui.Color.White, Terminal.Gui.Color.DarkGray);
     this.dialogBackground.Disabled = new Terminal.Gui.Attribute(Terminal.Gui.Color.Black, Terminal.Gui.Color.Black);
```

I would then instead generate code like:

```csharp
     this.dialogBackground = new Terminal.Gui.ColorScheme(
 new Terminal.Gui.Attribute(Terminal.Gui.Color.White, Terminal.Gui.Color.DarkGray),
new Terminal.Gui.Attribute(Terminal.Gui.Color.White, Terminal.Gui.Color.DarkGray),
new Terminal.Gui.Attribute(Terminal.Gui.Color.White, Terminal.Gui.Color.DarkGray),
new Terminal.Gui.Attribute(Terminal.Gui.Color.White, Terminal.Gui.Color.DarkGray),
new Terminal.Gui.Attribute(Terminal.Gui.Color.Black, Terminal.Gui.Color.Black));
```

**Comment by dodexahedron** [comment]
>Code DOM does not support initializer syntax

Yeah, it's fun with analyzers and code generation in general, since those have to be netstandard2.0-compatible.

You can work around it by including the missing interfaces and attributes, though. I've done exactly that in the analyzers and code generators I'm writing for TG right now. Most that are attributes are just metadata that may or may not even end up in the compiled assembly.

It sucks even outside the context of TG, but I'm feeling it here as well.

In general, any type that has an init-only property really should also have the required keyword set, should probably have a formal constructor or factory method available that ensures they are set, and any such constructors/methods should also have the [SetsRequiredMembers] attribute on them.
